metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfiguration.scala

```diff
@@ -0,0 +1,70 @@
+package scala.meta.internal.metals
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+
+import scala.meta.internal.bsp.BspServers
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.ClasspathSearch
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.MtagsResolver
+import scala.meta.internal.metals.ProgressTicks
+import scala.meta.internal.metals.Time
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.io.AbsolutePath
+
+/**
+ * Container class for metals language server and service configuration.
+ *   The configuration is mostly used to allow for custom behaviour in tests.
+ *   All the defaults are the instances used on the production.
+ */
+abstract class MetalsServerConfiguration {
+  val charset: Charset = StandardCharsets.UTF_8
```

Comment:
hmmm, I think we can make this just plain case class
```scala
final case class MetalsServerConfiguration(
  charset: Charset,
  buffers: Buffers,
  redirectSystemOut: Boolean,
  ...
)
```
and have default instance (similar like now) in companion object
```scala
object MetalsServerConfiguration {
  def default: MetalsServerConfiguration = MetalsServerConfiguration(
    charset = StandardCharsets.UTF_8,
    buffers = Buffers(),
    redirectSystemOut = true
    ...
  )
}
```
Using separate classes for prod and test purposes doesn't give us much.

Reply:
ok